guides/puppet-web-apps/html-template-guidelines/README.md

```diff
@@ -0,0 +1,157 @@
+
+For applications that have previously been developed without following these guidelines, take a look at the article [Convert Existing HTML Templates to starcounter-include](https://starcounter.io/making-apps-blendable/). To get a technical background, read the article [Layout compositions for HTML partials](https://starcounter.io/layout-compositions-html-partials/).
+
+### Guideline 1: Separate Functionality and Layout
+
+To make applications look great when running independently while also allowing them to visually blend with other applications, it is beneficial to separate the functionality and layout in the HTML template. This is easily accomplished using the `<template is="starcounter-composition">` element.
+
+The basic boilerplate of a Starcounter HTML template, which is created by adding a `Starcounter HTML template with dom-bind` file in Visual Studio, looks like this:
+
+```html
+<link rel="import" href="/sys/polymer/polymer.html">
+
+<template>
+    <template is="dom-bind">
+
+    </template>
+</template>
+```
+
+To separate functionality and layout in this html file, the element mentioned above, `<template is="starcounter-composition">` should be used. This element should contain the layout of the application while the `<template is="dom-bind">` should contain the functional elements. In code, this is how it looks:
+
+```html
+<link rel="import" href="/sys/polymer/polymer.html">
+
+<template>
+    <template is="dom-bind">
+        <!-- functional elements go here -->
+    </template>
+    <template is="starcounter-composition">
+        <!-- layout goes here -->
+    </template>
+</template>
+```
+
+A more in-depth explanation of `starcounter-composition` and why functionality and layout should be separated can be found in the [article linked above](https://starcounter.io/layout-compositions-html-partials/).
+
+### Guideline 2: Define the Functional Elements
+
+Functional elements are elements that either interact with the user visually, such as `<h1>`, `<p>` and `<span>` elements, or provides some kind of interaction between the user and the application, such as `<input>` and `<button>` elements. To start building the HTML template, these elements should be added to the `dom-bind` template as seen in Guideline 1.
+
+To have complete clarity in what counts as a functional elements, here are some examples:
+
+{% raw %}
+
+**Simple elements:**
+
+* `<button value="{{model.Submit::click}}">Submit</button>`
+* `<h1>Very Important Headline</h1>`
+* `<a href="/here/there">There</a>`
+* `<starcounter-include partial="{{model.MyPartial}}"></starcounter-include>`
+
+**Nested elements:**
+
+There are two kinds of nested elements that count as functional elements:

```

Comment:
> Who introduces such division of HTML Elements, and what for?

The distinction between nested and non-nested elements is clear for everyone. A nested element is simply an element that contains other elements.

In this part I'm showing that functional elements can be both nested and not. Although, that's not always the case, for example:
```html
<div>
  <h1>Something</h1>
  <h3>Else</h3>
</div>
```
This is a nested element, though it's not a functional element.

--- 

 > Only root elements of your HTML view are slotable. Put every meaningful element on root to make it slotable. This results in a perfectly blendable view.

I'm not sure if this actually makes any difference. Wouldn't "meaningful element" fill the same role as "functional element"? Or does "meaningful element" have a strict definition? "Semantic element" has a strict definition. Would that convey the same message?